From fd2d29e758f15d2dc79160096638f4b717cde103 Mon Sep 17 00:00:00 2001
From: borinskikh <causelesscause@gmail.com>
Date: Sun, 18 Apr 2021 15:55:06 +0300
Subject: [PATCH] patch-test

---
 config.def.h |  1 +
 tabbed.c     | 51 ++++++++++++++++++++++-----------------------------
 2 files changed, 23 insertions(+), 29 deletions(-)

diff --git a/config.def.h b/config.def.h
index defa426..c7e4967 100644
--- a/config.def.h
+++ b/config.def.h
@@ -12,6 +12,7 @@ static const char before[]      = "<";
 static const char after[]       = ">";
 static const char titletrim[]   = "...";
 static const int  tabwidth      = 200;
+static const int  tabheight	= 30;
 static const Bool foreground    = True;
 static       Bool urgentswitch  = False;
 
diff --git a/tabbed.c b/tabbed.c
index eafe28a..1218244 100644
--- a/tabbed.c
+++ b/tabbed.c
@@ -324,54 +324,47 @@ void
 drawbar(void)
 {
 	XftColor *col;
-	int c, cc, fc, width;
+	int c, cc, fc;
 	char *name = NULL;
 
+  dc.x = 0;
+	dc.w = tabwidth;
+
 	if (nclients == 0) {
-		dc.x = 0;
-		dc.w = ww;
 		XFetchName(dpy, win, &name);
 		drawtext(name ? name : "", dc.norm);
-		XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, ww, bh, 0, 0);
+		XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, tabwidth, wh, 0, 0);
 		XSync(dpy, False);
 
 		return;
 	}
 
-	width = ww;
-	cc = ww / tabwidth;
-	if (nclients > cc)
-		cc = (ww - TEXTW(before) - TEXTW(after)) / tabwidth;
+	cc = wh / tabheight;
+  if (nclients > cc)
+		cc -=2;
 
 	if ((fc = getfirsttab()) + cc < nclients) {
-		dc.w = TEXTW(after);
-		dc.x = width - dc.w;
-		drawtext(after, dc.sel);
-		width -= dc.w;
+		dc.y = wh - tabheight;
+		drawtext(after, dc.norm);
 	}
-	dc.x = 0;
+	dc.y = 0;
 
 	if (fc > 0) {
-		dc.w = TEXTW(before);
-		drawtext(before, dc.sel);
-		dc.x += dc.w;
-		width -= dc.w;
+		drawtext(before, dc.norm);
+		dc.y += tabheight;
 	}
 
 	cc = MIN(cc, nclients);
 	for (c = fc; c < fc + cc; c++) {
-		dc.w = width / cc;
 		if (c == sel) {
 			col = dc.sel;
-			dc.w += width % cc;
 		} else {
 			col = clients[c]->urgent ? dc.urg : dc.norm;
 		}
 		drawtext(clients[c]->name, col);
-		dc.x += dc.w;
-		clients[c]->tabx = dc.x;
+    dc.y += tabheight;
 	}
-	XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, ww, bh, 0, 0);
+	XCopyArea(dpy, dc.drawable, win, dc.gc, 0, 0, tabwidth, wh, 0, 0);
 	XSync(dpy, False);
 }
 
@@ -463,7 +456,7 @@ focus(int c)
 	if (c < 0 || c >= nclients)
 		return;
 
-	resize(c, ww, wh - bh);
+	resize(c, ww-tabwidth, wh);
 	XRaiseWindow(dpy, clients[c]->win);
 	XSetInputFocus(dpy, clients[c]->win, RevertToParent, CurrentTime);
 	sendxembed(c, XEMBED_FOCUS_IN, XEMBED_FOCUS_CURRENT, 0, 0);
@@ -589,9 +582,9 @@ getfirsttab(void)
 	if (sel < 0)
 		return 0;
 
-	cc = ww / tabwidth;
+	cc = wh / tabheight;
 	if (nclients > cc)
-		cc = (ww - TEXTW(before) - TEXTW(after)) / tabwidth;
+		cc -= 2;
 
 	ret = sel - cc / 2 + (cc + 1) % 2;
 	return ret < 0 ? 0 :
@@ -708,7 +701,7 @@ manage(Window w)
 		XEvent e;
 
 		XWithdrawWindow(dpy, w, 0);
-		XReparentWindow(dpy, w, win, 0, bh);
+		XReparentWindow(dpy, w, win, tabwidth, 0);
 		XSelectInput(dpy, w, PropertyChangeMask |
 		             StructureNotifyMask | EnterWindowMask);
 		XSync(dpy, False);
@@ -880,8 +873,8 @@ resize(int c, int w, int h)
 	XConfigureEvent ce;
 	XWindowChanges wc;
 
-	ce.x = 0;
-	ce.y = wc.y = bh;
+	ce.x = tabwidth;
+	ce.y = wc.y = 0;
 	ce.width = wc.width = w;
 	ce.height = wc.height = h;
 	ce.type = ConfigureNotify;
@@ -984,7 +977,7 @@ setup(void)
 	screen = DefaultScreen(dpy);
 	root = RootWindow(dpy, screen);
 	initfont(font);
-	bh = dc.h = dc.font.height + 2;
+	bh = dc.h = tabheight;
 
 	/* init atoms */
 	wmatom[WMDelete] = XInternAtom(dpy, "WM_DELETE_WINDOW", False);
-- 
2.31.1

